product_pack/migrations/12.0.1.0.0/pre-migration.py

```diff
@@ -0,0 +1,14 @@
+# Copyright 2019 Tecnativa - Ernesto Tejeda
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE product_template
+        SET pack_type = pack_price_type
+        """,
```

Comment:
Since the selection values has been fixed (misspelling error) we need to also add this here in order to do not lost the saved values.

* old "totalice_price" change to "totaliced_price"
* old "none_detailed_assited_price" to "none_detailed_assisted_price"

Reply:
Almost I miss, also the allow_modify_pack field has change from selection to boolean, Not sure if we need to add something here of automatically change to False if not set and True if one of the two old selections were selected.

Reply:
@rafaelbn @ernestotejeda 

yes, here is a funcional example. For the Detailed kind Pack Types, when the pack is added to a sale order and we save, we will see one line for the product pack and new sale order lines one for each pack components.

At the momento the salesman can edit the sale order lines and change the quantites of components of remove some of them. The Modificable Pack field is the one that was making this possible or not. If the field is set True then the pack can be edited and can be changed whaterver we want (quantities, remove products, change predefined discounts). If the field is set False then we avoid that the salesman can change the pack: the pack is forced to be sell as it is defined can not be modificable.

Modificable Pack field was before a selection field because we want to add the same funcionality in two scopes, backend and e-commerce. So a pack could be:

 * Not modificable: its forced to be sell as it defined.
 * Modificable only in backend: can be modificable by the internal user the salesman in the backend. This is if the salesman have any agreements with the customer and the company has the policy to let some flexibility to change the pack components/quantities/remove them.
 * Modificable in backend and e-commerce: the portal user can select a pack product and add it to their shop cart, and then change the quantities of the products (less/more) of the components and can also remove components from the pack.

In this case since the new product_pack module is a generic one that does not depends from sale or website, then this field was change to a simpler boolean (which it looks good to me).

Now I see that the modificable pack field was removed, I think this field should be leave in this module because can be used for sale_product_pack as well by the purchase_product_pack in the future.